keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceHistoryService.kt

```diff
@@ -92,53 +77,44 @@ class ResourceHistoryService(
     }
   }
 
-  private fun List<PersistentEvent>.isHappy(): Boolean {
+  private fun List<ResourceHistoryEvent>.isHappy(): Boolean {
     return first() is ResourceValid || first() is ResourceDeltaResolved
   }
```

Comment:
Couldn't the first element be an application paused/resumed event and still we're in a "happy" state?

I think you could potentially do `filterIsInstance<ResourceEvent>()` first (assuming I'm right about this).

Reply:
For the pause case, it could, but I short-circuit that case right above:
https://github.com/spinnaker/keel/blob/1a498d9759729a19066ecd280ae9b5b3d023e91f/keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusService.kt#L48-L54

In theory, if there's ever a pause record in the DB, there should be a matching event for that scope. In other words, if we ever get to evaluate other statuses, there should never be one of those at the top of the list.

As for the resume case, it has its own status, handled by the `isResumed()` function.

Reply:
So we cannot be both paused and happy?

Reply:
Nope, we can't. We only display that we're paused. when we resume, we will check again and then see if we're happy, at which point the status will change